src/lib/secrets/privkey_error.ml

```diff
@@ -12,8 +12,8 @@ let to_string : t -> string = function
       sprintf !"The key was corrupted: %s" (Error.to_string_hum e)
   | `Incorrect_password_or_corrupted_privkey ->
       "The password was incorrect, or the key is corrupted"
-  | `Cannot_open_file path ->
-      sprintf !"Cannot open file: %s" path
+  | `Cannot_open_file (path, e) ->
+      sprintf !"Cannot open file: %s. Error: %s" path (Unix.Error.message e)
```

Comment:
Don't think we need the `!` prefix on this (and other) print-format strings. That's needed for `{sexp: ...}`, which isn't used here.